horace_core/sqw/@Experiment/Experiment.m

```diff
@@ -164,55 +148,37 @@
             val=obj.expdata_;
         end
         function obj=set.expdata(obj, val)
-            if all(isempty(val))
-                if ~isa(val,'IX_experiment')
-                    obj.expdata_ = [];
-                    obj.runid_map_ = [];
-                    return;
-                end
-            end
-            if ~isa(val,'IX_experiment')
-                error('HORACE:Experiment:invalid_argument', ...
-                    'Sample must be one or an array of IX_experiment objects or empty. Actually it is: %s',...
-                    class(val))
-            end
-            obj.expdata_ = val(:)';
-            obj = build_runid_map_(obj);
-            if obj.do_check_combo_arg_
-                obj = check_combo_arg(obj);
-            end
-
+            %check and set the classes, containing main
+            %information about every run of the experiment
+            %
+            % Input should contain one or array of such classes
+            obj = check_and_set_expdata_(obj,val);
         end
         %
         function map = get.runid_map(obj)
             % deep copy handle class, to maintain consistent behaviour
             if isempty(obj.runid_map_)
                 map = [];
-            else
+            else % copy existing map as containers.Map is handle now.
                 map = containers.Map(obj.runid_map_.keys,obj.runid_map_.values);
             end
         end
         function obj = set.runid_map(obj,val)
-            % set runid or runid map and
-            %
-            if isa(val,'containers.Map')
-                obj.runid_map_ = val;
-                keys = val.keys;
-                keys = [keys{:}];
-            elseif isnumeric(val) && numel(val) == obj.n_runs
-                keys = val(:)';
-            else
-                error('HORACE:Experiment:invalid_argument', ...
-                    'input for runid_map should be map, defining connection between run-ids and headers(expdata), describing these runs or array of runid-s to set. It is %s', ...
-                    class(val))
-            end
-            obj = set_runids_map_and_synchonize_headers_(obj,keys);
-            if obj.do_check_combo_arg_
-                obj = check_combo_arg(obj);
-            end
-
+            % Check and set runid_map, connecting run-id, describing the
+            % experiment and the number of the experiment information
+            % header in the list of all experiment descriptors.
+            %
+            obj = set_runid_map_(obj,val);
         end
         %------------------------------------------------------------------
+        function nr = get.n_runs(obj)
+            nr = numel(obj.expdata_);
+            if nr == 1
+                if isempty(obj.expdata_(1))
+                    nr = 0;
+                end
+            end
```

Comment:
```suggestion
            if nr == 1 && isempty(obj.expdata_(1))
                nr = 0;
            end
```

Reply:
actually, this is old design. Cris have broken it